python/cudf/cudf/core/series.py

```diff
@@ -284,6 +284,15 @@ def __setitem__(self, key, value):
             to_dtype = np.result_type(value.dtype, self._frame._column.dtype)
             value = value.astype(to_dtype)
             if to_dtype != self._frame._column.dtype:
+                # Do not remove until pandas-3.0 support is added.
+                warnings.warn(
+                    f"Setting an item of incompatible dtype is deprecated "
```

Comment:
Additionally, what would happen when you have

```
ser = pd.Series()
ser[0] = 1
```

In pandas this casts object to int which is OK.

Reply:
We error when we try to set a non-string data to `object` dtype series. (Since cudf has no true object dtype, make perfect sense for pandas to cast in this case).

Reply:
True makes sense. I think it's OK that cudf is more strict here than pandas